sdk/metric/meter.go

```diff
@@ -200,6 +251,14 @@ func (m *meter) Float64ObservableGauge(name string, options ...metric.Float64Obs
 	return inst, nil
 }
 
+func validateInstrumentName(name string) error {
+	if !instrumentNameRe.MatchString(name) {
+		global.Warn("Invalid Instrument name. Instrument names must consist of 63 or fewer characters including alphanumeric, _, ., -, and start with a letter")
+		return errors.New("invalid instrument name")
```

Comment:
This returned error needs to be something a user can compare to. It is a "recoverable" error in the sense that a user should be able to ignore it and still use the SDK. They need a declared error to compare it to though.